enhancements/hypershift/support-disabling-image-registry-via-capability.md

```diff
@@ -0,0 +1,504 @@
+resulting in new pods in the management cluster. Additionally, the pull secrets
+controller (currently located in the openshift-controller-manager) will also be
+enabled by Hypershift, causing a pod redeployment.
+
+#### Standalone Clusters
+
+Standalone clusters already support capabilities - the changes in this
+enhancement do not affect this topology.
+
+#### Single-node Deployments or MicroShift
+
+The changes in this enhancement target Hypershift specific API and CLI.
+Single-node deployments and Microshift are unnaffected.
+
+### Implementation Details/Notes/Constraints
+
+With a capabilities framework in place, it's expected that more capabilities
+become available for disabling at install time.
+
+Currently in OpenShift Standalone all optional capabilities are enabled by default.
+In the future, that might change and some optional capabilities might come
+disabled by default.
+
+In that case, cluster service consumers might need a way to decide which
+of the disalbed optional capabilities they want to reenable at cluster install
+time. In this case, a `--enabled-capability` flag may be added to the Hypershift
+CLI, and an equivalent `EnbaledCapabilities` added to the `HostedCluster`
+resource.
+
+When extending the list of supported capabilities in Hypershift, it is crucial
+to use existing capability names from the CVO API. Standalone cluster
+capability names should exactly match the ones supported by Hypershift, although
+the set of supported capabilities is expected to differ.
+
+### Risks and Mitigations
+
+As users start to leverage the new capability, individual hosted control planes
+will start differing. The more optional capabilities are supported, the more one
+hosted control plane may differ from another. This may introduce support
+challenges (?).
+
+Managed services must implement support for the new `DisabledCapabilities` field
+in their own tooling before cluster service consumers can use them. Because of
+this, behavior may drift among different implementations, such as ARO and ROSA.
+
+### Drawbacks
+
+TBD
+
+The idea is to find the best form of an argument why this enhancement should
+_not_ be implemented.
+
+What trade-offs (technical/efficiency cost, user experience, flexibility,
+supportability, etc) must be made in order to implement this? What are the reasons
+we might not want to undertake this proposal, and how do we overcome them?
+
+Does this proposal implement a behavior that's new/unique/novel? Is it poorly
+aligned with existing user expectations?  Will it be a significant maintenance
+burden?  Is it likely to be superceded by something else in the near future?
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance,
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this?
+
+## Test Plan
```

Comment:
my list on e2e tests so far would be:
* TestCreateCluster [1] with image reg disabled
* TestCreateClusterV2 [2] with image reg disabled
* TestUpgradeControlPlane [3] with image reg disabled

the assertions they have are sufficient, we would just need to ensure the image registry operator is not showing up in the status objects after setting the capabilities 

[1] https://github.com/openshift/hypershift/blob/main/test/e2e/create_cluster_test.go#L1111
[2] https://github.com/openshift/hypershift/blob/main/test/e2e/create_cluster_test.go#L1165C6-L1165C25
[3] https://github.com/openshift/hypershift/blob/main/test/e2e/control_plane_upgrade_test.go#L16

Reply:
then we would need another integration (?) test to check the disable -> enable path is not allowed

Reply:
Thanks for this! I'll update this section. 

> then we would need another integration (?) test to check the disable -> enable path is not allowed

disable -> enable path is allowed, it's the opposite that cannot be done, i.e disabling an already enabled capability is not possible. But yes we do need a new test/assertion for that specifically.